metastats-domain/src/main/groovy/life/qbic/metastats/filter/FilterExperimentData.groovy

```diff
@@ -163,26 +164,43 @@ class FilterExperimentData implements FilterExperimentDataInput {
         boolean valid = false
```

Comment:
NIT: I usually try to avoid this approach, because it is error-prone. It is difficult to track the value of `valid` in your method and thus, you lose expressiveness and transparency. 

Moreover this variable states that your use case runs into a valid / invalid state at some point based on the input. So you can use class properties to track the state of your object (`this.invalidFileNames = true`). Think of the `Tell, don't ask principle`. 

I would recommend to overthink your two public methods the return boolean values. Why are they public?